### PR TITLE
Updates required for a stable deployment

### PR DESCRIPTION
**Pull request by viniciusdc**
## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

#### Text styling

- [ ] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [ ] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

### CODE REVIEW

File: playbook.yaml [hunk 1]
```diff
@@ -1,55 +1,55 @@
...
+      setup:
+      delegate_to: "{{ item }}"
+      delegate_facts: True
+      when: hostvars[item]['ansible_default_ipv4'] is not defined
+      with_items: "{{ groups['all'] }}"
+    - name: Copy files
+      include_tasks: tasks/copy_files.yaml
+      loop: '{{ ["all"] + group_names + [inventory_hostname_short] }}'
+      loop_control:
+        loop_var: myhost
```
Comment (aktech) [inline playbook.yaml:14]: are these changes necessary? If not I would suggest we avoid formatting changes in this PR to avoid noise and do that in a follow-up PR

Comment (aktech) [reply playbook.yaml:14]: I think that will be great and will be good for posterity, so that we can come back and see what did it took to make it work in 2024 with the latest ubuntu 2204.

**Comment by viniciusdc** [comment]
Hi @aktech just refactored the changes to avoid the linting. May I add your changes from amit-deploy here as well?

File: roles/jupyterhub/templates/environments/jupyterhub.yaml [hunk 2]
```diff
@@ -2,17 +2,18 @@ name: jupyterhub
...
   - conda-forge
 dependencies:
-  - pip
-  - jupyterhub{{ jupyterhub_version }}
-  - oauthenticator
-  - jupyterhub-idle-culler
-  - jupyterhub-systemdspawner
-  - python-keycloak
+  - pip==21.1.2
+  - jupyterhub==2.3.0
```
Comment (aktech) [inline roles/jupyterhub/templates/environments/jupyterhub.yaml:6]: any reason for `{{ jupyterhub_version }}` -> `2.3.0`

Comment (viniciusdc) [reply roles/jupyterhub/templates/environments/jupyterhub.yaml:6]: When I was reviewing the environments to make them work, I checked an old Qhub deployment I had running on that version. No need to keep this one

**Comment by aktech** [comment]
> Hi @aktech just refactored the changes to avoid the linting. May I add your changes from amit-deploy here as well?

I have added upgrade changes to it as well, which is not ready for merge. I would say skip that for now.

File: roles/jupyterhub/templates/environments/jupyterhub.yaml (hunk 2 above)
Comment (viniciusdc) [reply roles/jupyterhub/templates/environments/jupyterhub.yaml:6]: the reason I moved from the env var to hardcoded as for testing, we can have the env var back and just update the version in there

Comment (viniciusdc) [reply roles/jupyterhub/templates/environments/jupyterhub.yaml:6]: oh, I remembered now, it was set to `>=2.3.0` and it was getting the latest which was breaking the other envs, that's why I pinned (moved from var to hardcoded for the reason above)

File: roles/jupyterhub/templates/environments/jupyterlab.yaml [hunk 3]
```diff
@@ -2,22 +2,51 @@ name: jupyterlab
...
+  - flake8
+  - sphinx
+  - pytest-cov
+  - black
+  - isort
+  - importnb
+
   - pip:
-      # conda-store (mamba-org/gator)
-      - https://github.com/mamba-org/gator/archive/conda-store-integration.tar.gz
```
Comment (aktech) [inline roles/jupyterhub/templates/environments/jupyterlab.yaml:20]: Why was this removed?

Comment (aktech) [reply roles/jupyterhub/templates/environments/jupyterlab.yaml:20]: nevermind, we don't use it anyway

**Review by aktech — APPROVED** [review]
